frontend/src/components/DataProvider/index.js

```diff
@@ -1,12 +1,16 @@
 import { $ } from '@/commons/utils/query-selector';
+import Notifications from '@/components/Notifications/index';
 import Todos from '@/components/Todos/index';
 import { getTodos } from '@/libs/api';
 import Component from '@/libs/Component';
 
 class DataProvider extends Component {
-  constructor($container, initialState) {
+  constructor($container, initialState, setNotificationsOpenState, $notificationsComponent) {
     super($container, initialState);
 
+    this.setNotificationsOpenState = setNotificationsOpenState;
+    this.$notificationsComponent = $notificationsComponent;
+
```

Comment:
```suggestion
  constructor($container, initialState, setNotificationsOpenState) {
    super($container, initialState);

    this.setNotificationsOpenState = setNotificationsOpenState;
    this.$notificationsComponent = null;
```

이런 식으로 초기화해서 사용하는 건 어떨까요?

DataProvider 생성자를 호출하는 app.js에서 `$notificationsComponent` 가 undefined로 넘어오는 것으로 예상이 되네요!

그 부분도 같이 조정하면 좋을 것 같습니다!!!

Reply:
불필요하게 사용하지 않는 컴포넌트는 지우고, 혁진님이 말씀해주신 것처럼 초기화해서 사용하도록 수정했습니다.
감사합니다  😀